devicetypes/smartthings/zigbee-motion-sensor-light.src/led-cpx-light.groovy

```diff
@@ -0,0 +1,166 @@
+	}
+}
+
+private getMOTION_CLUSTER_VALUE() { 0x0406 }
+private getMOTION_STATUS_VALUE() { 0x0000 }
+private getON_OFF_CLUSTER_VALUE() { 0x0006 }
+
+// Parse incoming device messages to generate events
+def parse(String description) {
+	def event = zigbee.getEvent(description)
+	def zigbeeMap = zigbee.parseDescriptionAsMap(description)
+	if (event) {
+		if (!(event.name == "level" && event.value == 0)) {
+			sendEvent(event)
+		}
+	} else if (zigbeeMap.cluster == MOTION_CLUSTER_VALUE && zigbeeMap.attrId == MOTION_STATUS_VALUE) {
+		def childDevice = getChildDevices()?.find {
+			it.device.deviceNetworkId == "${device.deviceNetworkId}:1"
+		}
+		zigbeeMap.value = zigbeeMap.value.endsWith("01") ? "active" : "inactive"
+		zigbeeMap.name = "motion"
+		childDevice.sendEvent(zigbeeMap)
+	} else {
+		def cluster = zigbee.parse(description)
+		if (cluster && cluster.clusterId == ON_OFF_CLUSTER_VALUE && cluster.command == 0x07) {
+			if (cluster.data[0] == 0x00) {
+				sendEvent(name: "checkInterval", value: 60 * 12, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
+			}
+		}
+	}
+}
+
+def off() {
+	zigbee.off()
+}
+
+def on() {
+	zigbee.on()
+}
+
+def ping() {
+	return zigbee.onOffRefresh()
+}
+
+def refresh() {
+	zigbee.onOffRefresh() +
+		zigbee.levelRefresh() +
+		zigbee.colorTemperatureRefresh() +
+		zigbee.onOffConfig(0, 300) +
+		zigbee.levelConfig()
+}
+
+def configure() {
+	def cmds = delayBetween([
+		sendEvent(name: "checkInterval", value: 2 * 10 * 60 + 1 * 60, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID]),
+		zigbee.configureReporting(MOTION_CLUSTER_VALUE, MOTION_STATUS_VALUE, 0x18, 30, 600, null),
+		zigbee.onOffRefresh(),
+		zigbee.levelRefresh(),
+		zigbee.colorTemperatureRefresh(),
+		zigbee.onOffConfig(0, 300),
+		zigbee.levelConfig()
+	])
+	return cmds
+}
+
+def setColorTemperature(value) {
+	value = value as Integer
+	def tempInMired = Math.round(1000000 / value)
+	def finalHex = zigbee.swapEndianHex(zigbee.convertToHexString(tempInMired, 4))
+	
+	List cmds = []  
+	cmds << zigbee.command(COLOR_CONTROL_CLUSTER, MOVE_TO_COLOR_TEMPERATURE_COMMAND, "$finalHex 0000")
+	cmds << zigbee.readAttribute(COLOR_CONTROL_CLUSTER, ATTRIBUTE_COLOR_TEMPERATURE)
+	cmds
+}
+
+def installed() {
+	addChildSensor()
+	return
+}
+
+def addChildSensor() {
+	def componentLabel
+	if (device.displayName.endsWith(' Light') || device.displayName.endsWith(' light')) {
+		componentLabel = "${device.displayName[0..-6]} Motion sensor"
+	} else {
+		componentLabel = "$device.displayName Motion sensor"
+	} 
+	try {
+		String dni = "${device.deviceNetworkId}:1"
+		addChildDevice("ITM CPX Motion sensor child", dni, device.hub.id, [completedSetup: true, label: "${componentLabel}", isComponent: false])
+	} catch (e) {
+		log.warn "Failed to add ITM Motion sensor Controller - $e"
+	}
+	
+	def childDevice = getChildDevices()?.find {
+		it.device.deviceNetworkId == "${device.deviceNetworkId}:1"
+	}
```

Comment:
you can also just use the object returned by `addChildDevice` instead of calling this find immediately after

Reply:
Bump. 
```
def childDevice = addChildDevice("ITM CPX Motion sensor child", dni, device.hub.id, [completedSetup: true, label: "${componentLabel}", isComponent: false])
childDevice.sendEvent(name: "motion", value: "inactive")
```